### PR TITLE
fix probe promcfg

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

Fixes prometheus config for Probe.

1. If only ProbeSelector is set, PodMonitorSelector and ServiceMonitorSelector not set, it will do nothing.
2. Relabelings are not added for Probe.

### CODE REVIEW

File: pkg/prometheus/promcfg.go [hunk 1]
```diff
@@ -747,22 +762,6 @@ func (cg *configGenerator) generateProbeConfig(
 			cfg = append(cfg, cg.generateK8SSDConfig(selectedNamespaces, apiserverConfig, basicAuthSecrets, kubernetesSDRoleIngress))
 		}
 
-		// Relabelings for prober.
```
Comment (brancz) [inline pkg/prometheus/promcfg.go:750]: why would this not be needed for ingress targets?

Comment (yeya24) [reply pkg/prometheus/promcfg.go:750]: Thanks for the review, I will add it back.

**Comment by yeya24** [comment]
I am not sure about the current code, why static configs don't have relabeling configs? Is it a bug or expected?

This is one expected result in the unit test.

``` yaml
global:
  evaluation_interval: 30s
  scrape_interval: 30s
  external_labels:
    prometheus: default/test
    prometheus_replica: $(POD_NAME)
rule_files: []
scrape_configs:
- job_name: default/testprobe1
  honor_timestamps: true
  metrics_path: /probe
  scheme: http
  params:
    module:
    - http_2xx
  static_configs:
  - targets:
    - prometheus.io
    - promcon.io
    labels:
      static: label
  relabel_configs: []
alerting:
  alert_relabel_configs:
  - action: labeldrop
    regex: prometheus_replica
  alertmanagers: []
```

**Comment by brancz** [comment]
Static targets shouldn't have relabel configs, it doesn't make sense, they should just have labels, which can already be configured.

**Comment by brancz** [comment]
Thanks! :)

**Comment by yeya24** [comment]
> Static targets shouldn't have relabel configs, it doesn't make sense, they should just have labels, which can already be configured.

Sorry, I just saw your reply. Do you mean the relabel configs other than those for Blackbox exporter? Or including the Blackbox exporter relabel configs?

I add these relabel configs to static config as well. If this is wrong, I can open another pr to fix it.

```
		relabelings = append(relabelings, []yaml.MapSlice{
			{
				{Key: "source_labels", Value: []string{"__address__"}},
				{Key: "target_label", Value: "__param_target"},
			},
			{
				{Key: "source_labels", Value: []string{"__param_target"}},
				{Key: "target_label", Value: "instance"},
			},
			{
				{Key: "target_label", Value: "__address__"},
				{Key: "replacement", Value: m.Spec.ProberSpec.URL},
			},
		}...)
```

**Comment by brancz** [comment]
This looks good now. But to make 100% sure, would you mind adding a full e2e test for static and ingress discovered targets? Just so we can be totally sure it works end to end.

**Comment by yeya24** [comment]
> This looks good now. But to make 100% sure, would you mind adding a full e2e test for static and ingress discovered targets? Just so we can be totally sure it works end to end.

Yes, I can add E2E tests in next prs.

**Comment by brancz** [comment]
Awesome! :tada:

**Pull request merged**